CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+
+* Make your changes in a new git branch:
+
+    ```shell
+    git checkout -b my-fix-branch master
+    ```
+
+* Create your patch commit, **including appropriate test cases**.
+* Follow our Coding Rules.
+* If the changes affect public APIs, change or add relevant documentation.
+* Commit your changes using a descriptive commit message that follows our
+  commit message conventions. Adherence to the
+  commit message conventions is required.
+
+    ```shell
+    git commit -a
+    ```
+  Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+
+* Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
+    ```
+
+* First the commit has to pass the continous integration test. We choose drone as our ci tool.
+
+* If you find that the drone integration has failed, look into the logs on drone to find out
+if your changes caused test failures etc.
+
+* If we suggest changes, then:
+
+  * Make the required updates
+  * Commit your changes to your branch (e.g. `my-fix-branch`).
+  * Push the changes to your GitHub repository (this will update your Pull Request).
+
+    You can also amend the initial commits and force push them to the branch.
+
+    ```shell
+    git rebase master -i
+    git push origin my-fix-branch -f
+    ```
+
+    This is generally easier to follow, but seperate commits are useful if the Pull Request contains
+    iterations that might be interesting to see side-by-side.
+
+That's it! Thank you for your contribution!
+
+#### After your pull request is merged
+
+After your pull request is merged, you can safely delete your branch and pull the changes
+from the main (upstream) repository:
+
+* Delete the remote branch on GitHub either through the GitHub web UI or your local shell as follows:
+
+    ```shell
+    git push origin --delete my-fix-branch
+    ```
+
+* Check out the master branch:
+
+    ```shell
+    git checkout master -f
+    ```
+
+* Delete the local branch:
+
+    ```shell
+    git branch -D my-fix-branch
+    ```
+
+* Update your master with the latest upstream version:
+
+    ```shell
+    git pull --ff upstream master
+```
+
+## Coding Rules
+
+We follow the PEP-8 styleguide. There are some tools out there that formats your code PEP-8 conform.
+E.g. for [visual studio code](https://marketplace.visualstudio.com/items?itemName=himanoa.Python-autopep8)
+
+If you work on the tensorflow part of the project also follow these [guidelines](https://danijar.com/structuring-your-tensorflow-models/).
+
+## Git Commit Guidelines
+
+We have very precise rules over how our git commit messages can be formatted.  This leads to **more
+readable messages** that are easy to follow when looking through the **project history**.
+
+### Commit Message Format
+Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+format that includes a **type**, a **scope** and a **subject**:
+
+```
```

Comment:
This example is broken in the rich diff. A simple example like "enh(contributing): created Contributing.md and mall cahnges in Readme" is enough i think

Reply:
beneath the shown code... still dont know how to do this correctly